### DISCUSSION TITLE
Roadmap for 2022

### DISCUSSION THREAD

**Discussion by ppstacy** [uber/causalml#451] [Ideas]
Hi everyone,

Hope you have a fulfilling 2021 and a great start to 2022. Thanks for all the contributions and attention to CausalML! Since our first release in Aug 2019, CausalML has surpassed [637K downloads](https://pepy.tech/project/causalml) and [2,500 stars](https://github.com/uber/causalml/stargazers) :tada:. While we are celebrating, it’s also a good time to plan for the future and discuss what can continue contributing to the community and to the industry.

Below are some of the key focusing areas the CausalML dev team is considering and please comment below anything if you think good to have or want the CausalML to support for. We can discuss and brainstorm with the team.

 **Scalability**
- Scale the current estimators with frameworks for example: [Ray](https://github.com/ray-project/ray), TensorFlow, PyTorch, [JAX](https://github.com/google/jax) etc.
- Scale with the large experimentation datasets 

 **Algorithms**: more and more Causal Inference algorithms have been developed especially the deep learning/neural nets areas that CausalML could consider support for.

Please comment or just share ideas below. Thanks and keep safe!!

**Comment by ppstacy** [comment]
I will start with some interesting papers from NeurIPS 2021: 
[Causal-BALD](https://papers.nips.cc/paper/2021/hash/ffa4eb0e32349ae57f7a0ee8c7cd7c11-Abstract.html)
[SCP](https://papers.nips.cc/paper/2021/hash/c793b3be8f18731f2a4c627fb3c6c63d-Abstract.html)
[SurvITE](https://papers.nips.cc/paper/2021/hash/e0eacd983971634327ae1819ea8b6214-Abstract.html)
[Estimating the Long-Term Effects of Novel Treatments](https://papers.nips.cc/paper/2021/hash/16fa2b0294e410b2551c3bf6965c0853-Abstract.html)
[Double Machine Learning Density Estimation for Local Treatment Effects with Instruments ](https://papers.nips.cc/paper/2021/hash/b61a560ed1b918340a0ddd00e08c990e-Abstract.html)

**Comment by huigangchen** [comment]
On the A/B test regression adjustment, there has been a stream of interesting papers since Freedman's surprising result (2008) that shows regression adjustment can be biased in an A/B test. AFAIK the current state-of-art paper is 

[No-harm calibration for generalized Oaxaca-Blinder estimators
](https://arxiv.org/abs/2012.09246#:~:text=This%20recovers%20the%20usual%20regression,as%20logistic%20and%20Poisson%20regression.)

And I think this can be a good direction for our package to move into the space of experimentation platform.

**Comment by vincewu51** [comment]
For using deep learning to help causal inference, I found this review ([Deep Learning of Potential Outcomes](https://arxiv.org/pdf/2110.04442.pdf)) to be a good starting point.

**Reply by vincewu51** [reply]
https://github.com/kochbj/Deep-Learning-for-Causal-Inference
